packages/viz/src/lib/sources/DOSource.ts

```diff
@@ -79,28 +129,59 @@ export class DOSource extends Source {
     return parseFetchJSON(r);
   }
 
-  public async getLayerProps(): Promise<DataObservatoryLayerProps> {
-    const vizURL = `${this._baseURL}/visualization`;
-    const { apiKey } = this._credentials;
+  private async _getGeography(geographyID: string): Promise<Geography> {
+    const url = `${this._baseURL}/metadata/geographies/${geographyID}`;
+    const r = await fetch(url);
+    return parseFetchJSON(r);
+  }
 
+  public async init(): Promise<boolean> {
     // Get geography from metadata
-    const variable = await this._getVariable();
+    const variable = await this._getVariable(this._variable);
     const dataset = await this._getDataset(variable.dataset_id);
+    const geography = await this._getGeography(dataset.geography_id);
+
+    this._model = { dataset, variable, geography };
+    this.isInitialize = true;
+    return this.isInitialize;
+  }
 
-    const geography = dataset.geography_id;
+  public getLayerProps(): DOLayerProps {
+    if (!this.isInitialize || this._model === undefined)
+      throw new Error('To run this method you need to first call to init()');
+
+    const vizURL = `${this._baseURL}/visualization`;
+    const { apiKey } = this._credentials;
+
+    // Get geography from metadata
+    const geography = this._model.dataset.geography_id;
 
     const geographiesURLTemplate = `${vizURL}/geographies/${geography}/{z}/{x}/{y}.mvt?api_key=${apiKey}`;
     const dataURLTemplate = `${vizURL}/variables/{z}/{x}/{y}.json?variable=${this._variable}&api_key=${apiKey}`;
-    const geometryType = 'MultiPolygon';
 
-    return { geographiesURLTemplate, dataURLTemplate, geometryType };
+    return { type: 'TileLayer', geographiesURLTemplate, dataURLTemplate };
   }
 
-  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-  // @ts-ignore
-  // eslint-disable-next-line
-  public async getMetadataForField(field: string): Promise<Metadata> {
-    throw new Error('Method not implemented.');
+  public getFieldStats(field: string): NumericFieldStats {
+    if (!this.isInitialize || this._model === undefined) {
+      throw new Error('To run this method you need to first call to init()');
```

Comment:
We can use here specific types of CartoError. We can add them progressively...

Reply:
Yes, I didn't realize the existence of CartoError when I wrote this part. Let's do it